### PR TITLE
Fix compilation warnings around CAStore

### PR DESCRIPTION
**Pull request by whatyouhide**
* Mix was complaining when compiling Mint as a dependency if `:castore` was not in the dependencies too. Solved this by not using `CAstore` as a compile-time module reference, but making it dynamic instead.
* **Unrelated refactor**: I just used `Keyword.put_new/3` and `Keyword.put_new_lazy/3` in favor of some `if` checks in `Mint.Core.Transport.SSL`.

### CODE REVIEW

File: lib/mint/core/transport/ssl.ex [hunk 1]
```diff
@@ -565,11 +555,14 @@ defmodule Mint.Core.Transport.SSL do
   end
 
   defp add_cacerts(opts) do
-    if Keyword.has_key?(opts, :cacertfile) || Keyword.has_key?(opts, :cacerts) do
+    if Keyword.has_key?(opts, :cacertfile) or Keyword.has_key?(opts, :cacerts) do
```
Comment (whatyouhide) [inline lib/mint/core/transport/ssl.ex:558]: Unrelated refactor, strict `or` since both sides are boolean.

File: lib/mint/core/transport/ssl.ex [hunk 2]
```diff
@@ -565,11 +555,14 @@ defmodule Mint.Core.Transport.SSL do
...
       opts
     else
       raise_on_missing_castore!()
-      Keyword.put(opts, :cacertfile, CAStore.file_path())
+      # Use a dynamic module name to avoid compilation warnings when Mint is used as
+      # a dependency without :castore.
+      castore_mod = CAStore
+      Keyword.put(opts, :cacertfile, castore_mod.file_path())
```
Comment (whatyouhide) [inline lib/mint/core/transport/ssl.ex:565]: Main fix.

**Review by ericmj — APPROVED** [review]
